python/lsst/obs/fiberspectrograph/translator.py

```diff
@@ -0,0 +1,214 @@
+
+    name = "FiberSpectrograph"
+    """Name of this translation class"""
+
+    supported_instrument = "FiberSpec"
+    """Supports the Rubin calibration fibre spectrographs."""
+
+    default_search_path = os.path.join(getPackageDir("obs_fiberspectrograph"), "corrections")
+    """Default search path to use to locate header correction files."""
+
+    default_resource_root = os.path.join(getPackageDir("obs_fiberspectrograph"), "corrections")
+    """Default resource path root to use to locate header correction files."""
+
+    DETECTOR_MAX = 1
+
+    _const_map = {
+        "detector_num": 0,
+        "detector_name": "0",
+        "exposure_group": None,
+        "object": None,
+        "physical_filter": "empty",
+        "detector_serial": "0xdeadbeef",
+        "detector_group": "None",
+        "relative_humidity": None,
+        "pressure": None,
+        "temperature": None,
+        "focus_z": None,
+        "boresight_airmass": None,
+        "boresight_rotation_angle": None,
+        "tracking_radec": None,
+        "telescope": SIMONYI_TELESCOPE,
+        "observation_type": "spectrum",  # IMGTYPE is ''
+    }
+    """Constant mappings"""
+
+    _trivial_map = {
+        "observation_id": "OBSID",
+        "science_program": ("PROGRAM", dict(default="unknown")),
+    }
+    """One-to-one mappings"""
+
+    @classmethod
+    def can_translate(cls, header, filename=None):
+        """Indicate whether this translation class can translate the
+        supplied header.
+
+        Parameters
+        ----------
+        header : `dict`-like
+            Header to convert to standardized form.
+        filename : `str`, optional
+            Name of file being translated.
+
+        Returns
+        -------
+        can : `bool`
+            `True` if the header is recognized by this class. `False`
+            otherwise.
+        """
+
+        return "INSTRUME" in header and header["INSTRUME"] in ["FiberSpectrograph.Broad"]
+
+    @cache_translation
+    def to_instrument(self):
+        return "FiberSpec"
+
+    @cache_translation
+    def to_datetime_begin(self):
+        self._used_these_cards("DATE-BEG")
+        return Time(self._header["DATE-BEG"], scale="tai", format="isot")
+
+    @cache_translation
+    def to_observing_day(self):
+        """Return the day of observation as YYYYMMDD integer.
+
+        Returns
+        -------
+        obs_day : `int`
+            The day of observation.
+        """
+        date = self.to_datetime_begin()
+        date -= self._ROLLOVER_TIME
+        return int(date.strftime("%Y%m%d"))
+
+    @cache_translation
+    def to_observation_counter(self):
+        """Return the sequence number within the observing day.
+
+        Returns
+        -------
+        counter : `int`
+            The sequence number for this day.
+        """
+        if self.is_key_ok("OBSID"):
+            self._used_these_cards("OBSID")
+            return int(self._header["OBSID"])
```

Comment:
If this resets at the start of each day there is going to be a problem.

Reply:
Fixed in the new files